ecommerce/extensions/basket/utils.py

```
@@ -91,3 +87,70 @@ def get_basket_switch_data(product):
             partner_sku = stock_record.partner_sku
             break
     return switch_link_text, partner_sku
+
+
+def attribute_cookie_data(basket, request):
+    try:
+        referral = _referral_from_basket_site(basket, request.site)
+
+        _record_affiliate_basket_attribution(referral, request)
+        _record_utm_basket_attribution(referral, request)
+
+        # Save the record if any attribution attributes are set on it.
+        if any([getattr(referral, attribute) for attribute in Referral.ATTRIBUTION_ATTRIBUTES]):
+            referral.save()
+        # Clean up the record if no attribution attributes are set and it exists in the DB.
+        elif referral.pk:
+            referral.delete()
+        # Otherwise we can ignore the instantiated but unsaved referral
+
+    # Don't let attribution errors prevent users from creating baskets
+    except Exception:  # pylint: disable=broad-except
```

You could just use a bare except here.